tests/Doctrine/Tests/ORM/Hydration/ScalarColumnHydratorTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Hydration;
+
+use Doctrine\ORM\Exception\MultipleSelectorsFoundException;
+use Doctrine\ORM\Internal\Hydration\ScalarColumnHydrator;
+use Doctrine\ORM\Query\ResultSetMapping;
+use Doctrine\Tests\Mocks\ResultMock;
+use Doctrine\Tests\Models\CMS\CmsUser;
+
+use function count;
+use function is_array;
+use function sprintf;
+
+class ScalarColumnHydratorTest extends HydrationTestCase
+{
+    /**
+     * Select u.id from CmsUser u
+     */
+    public function testEmptyResultTest(): void
+    {
+        $rsm = new ResultSetMapping();
+
+        $rsm->addEntityResult(CmsUser::class, 'u');
+        $rsm->addFieldResult('u', 'u__id', 'id');
+
+        $stmt     = new ResultMock($emptyResultSet = []);
+        $hydrator = new ScalarColumnHydrator($this->entityManager);
+        $result   = $hydrator->hydrateAll($stmt, $rsm);
+
+        $this->assertTrue(is_array($result));
```

Comment:
```suggestion
        $this->assertIsArray($result));
```

The same with the other tests of ScalarColumnHydratorTest.

Reply:
Updated